homeassistant/components/climate/broadlink.py

```diff
@@ -0,0 +1,414 @@
+import logging
+import json
+import voluptuous as vol
+import homeassistant.helpers.config_validation as cv
+from socket import timeout
+from homeassistant.components.climate import ClimateDevice, \
+    SUPPORT_TARGET_TEMPERATURE,  PLATFORM_SCHEMA, \
+    SUPPORT_OPERATION_MODE, STATE_AUTO
+from homeassistant.const import TEMP_CELSIUS, ATTR_TEMPERATURE, CONF_HOST,\
+    CONF_MAC, CONF_NAME, STATE_ON, STATE_OFF
+
+"""
+Support for Chinese wifi thermostats (Floureon, Beok, Beca Energy)
+configuration.yaml
+climate:
+  - platform: broadlink
+    name: xxx
+    mac: xxxx
+    host: xxxx
+"""
+
+_LOGGER = logging.getLogger(__name__)
+DOMAIN = 'broadlink'
+REQUIREMENTS = ['broadlink==0.9.0']
+DEPENDENCIES = []
+
+POWER_ON = 1
+POWER_OFF = 0
+AUTO = 1
+MANUAL = 0
+CONF_MODE_LIST = 'modes'
+CONF_MIN_TEMP = 'min_temp'
+CONF_MAX_TEMP = 'max_temp'
+CONF_ADVANCED_CONFIG = 'advanced_config'
+CONF_SCHEDULE_WEEKDAY = 'schedule_week_day'
+CONF_SCHEDULE_WEEKEND = 'schedule_week_end'
+CONF_WEEKDAY = "weekday"
+CONF_WEEKEND = "weekend"
+
+# Validation of the user's configuration
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_MAC): cv.string,
+    vol.Required(CONF_NAME): cv.string,
+    vol.Optional(CONF_MIN_TEMP, default=5): cv.positive_int,
+    vol.Optional(CONF_MAX_TEMP, default=35): cv.positive_int,
+    vol.Optional(CONF_ADVANCED_CONFIG,
+                 default='{"loop_mode": "0", '
```

Comment:
Don't mix json and yaml. Probably drop the advanced config altogether.

Reply:
I may can split the advance config json into multiple yaml entry, but i thinks that a bit overkill...
May be not setting a default value can fix it ?